src/pages/calendar/components/Requisites.tsx

```diff
@@ -0,0 +1,192 @@
+import { Box, Text, Divider, ListItem, Skeleton, UnorderedList } from '@chakra-ui/react';
+import { Link } from 'react-router-dom';
+
+import { KualiCourse, NestedPreCoRequisites, GetCourseDetails, Term } from 'lib/fetchers';
+import { useSessionStorage } from 'lib/hooks/storage/useSessionStorage';
+import { getCurrentTerm } from 'lib/utils/terms';
+
+// Based on the nested info in the requisite, display the element necessary
+export function DisplayRequirement(
+  req: string | NestedPreCoRequisites | KualiCourse,
+  indentationLevel: number = 1
+): JSX.Element {
+  // Grab the current term
+  const currTerm = useSessionStorage('user:term', getCurrentTerm())[0] as Term;
+
+  // If its just a string, display it. Eg. "or permission from the department"
+  if (typeof req === 'string') {
+    return (
+      <UnorderedList>
+        <ListItem title="Requisite" style={{ marginLeft: `${indentationLevel * 40}px` }}>
+          {req}
+        </ListItem>
+      </UnorderedList>
+    );
+    // If there's a quantity, then we have a list of requisites to display
+  } else if ('quantity' in req) {
+    //Then it is a NestedPreCoRequisite
+    const reqs = nestedReq(req);
+    if (req.coreq) {
+      return (
+        <Box>
+          {req.quantity && ( // If all are required, it doesn't get displayed
+            <UnorderedList>
+              <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                Completed or currently enrolled in {req.quantity} of:{' '}
+              </ListItem>
+            </UnorderedList>
+          )}
+          {req.grade && (
+            <UnorderedList>
+              <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                Earn a minimum grade of {req.grade} in each of:
+              </ListItem>
+            </UnorderedList>
+          )}
+          {reqs.map((r) => DisplayRequirement(r, indentationLevel + 1))}{' '}
```

Comment:
```suggestion
          {reqs.map((r) => DisplayRequirement(r, indentationLevel + 1))}
```
same - sometimes these just get added by the linter, but keep if it's being used